edge-apps/helpers/tfl/bus-stop-lookup/script.js

```diff
@@ -1,85 +1,566 @@
+/* eslint-disable no-unused-vars */
+
 /* global Papa */
 
 document.addEventListener('DOMContentLoaded', function () {
+  // Create initial page structure to avoid timing issues
+  setupPageStructure()
+
+  // Get references to key elements after they're created
   const searchInput = document.getElementById('search')
   const tableHead = document.querySelector('thead tr')
   const tableBody = document.querySelector('tbody')
+  const loadingIndicator = document.getElementById('loading')
+  const paginationContainer = document.getElementById('pagination')
+  const filterContainer = document.getElementById('filters')
+
+  // Create tooltip for copy feedback
+  const tooltip = document.createElement('div')
+  tooltip.className = 'copy-tooltip'
```

Comment:
Prefer using tooltip.classList.add / remove to add remove class so as to preserve existing classes. Does not matter here though.